libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -1272,26 +1272,21 @@ void AP_InertialSensor::update(void)
     wait_for_sample();
 
     if (!_hil_mode) {
-        for (uint8_t i=0; i<INS_MAX_INSTANCES; i++) {
-            // mark sensors unhealthy and let update() in each backend
-            // mark them healthy via _publish_gyro() and
-            // _publish_accel()
-            _gyro_healthy[i] = false;
-            _accel_healthy[i] = false;
-            _delta_velocity_valid[i] = false;
-            _delta_angle_valid[i] = false;
-        }
+        // mark sensors unhealthy and let update() in each backend
+        memset(_gyro_healthy,           0, sizeof(_gyro_healthy[0]) * INS_MAX_INSTANCES);
```

Comment:
I wonder if `ARRAY_SIZE(_gyro_healthy)` would be better than INS_MAX_INSTANCES?

Reply:
Theoretically one could do that, but it is all the same at the moment.